### ISSUE TITLE
Deprecate Notice in favor of Alert

### ISSUE THREAD

**Issue by henningmu** [Doist/reactist#702]
@gnapse I'm unclear on the difference between our `Notice` component and the `Alert` component. Shouldn't one be enough? Checking the Product Library Figma, there is also only the Alert and no concept of a Notice 🤔

**Comment by gnapse** [comment]
This may be an issue about lack of documentation. The lack of a Figma design is a form of lack of documentation, but I'm referring to a lack of documentation in the code itself, in the storybook examples, for instance.

There are uses of `<Notice />` in both products, and that's the ultimate "source of truth". It has new uses added as recently as when we worked on Todoist's 2FA. So ultimately, this is something our Design team should answer.

@Doist/design-product-hero do we really need [the `Notice` component](https://doist.github.io/reactist/?path=/story/design-system-notice--playground)? What is its place in contrast with [the `Alert` component](https://doist.github.io/reactist/?path=/story/design-system-alert--playground)?

Some comments on my side from what I think about having them both, and what their differences and similarities are:

- They fulfill very similar purposes. `Alert` has more weight, and draws more attention than notice.
- `Alert` makes it possible that the user can close it (if optionally supported, not all `Alerts` allow this). `Notice` does not offer this feature. A `Notice` only goes away when the condition that makes it appear goes away.
- We already have some other UI elements that are visually similar to `Notice`, but in the context of a larger component. Most notable, and possibly the only example so far, is [the message feature in our field components](https://share.cleanshot.com/4JYYsj). This is not something that I say to support the existence of `Notice`. This is to say that maybe that's one acceptable use, but that all standalone instances of `Notice` could/should be replaced by `Alert`.

## Examples

A couple of examples for how each component looks like:

<table>
<thead><tr><th>

`Notice`

</th><th>

`Alert`

</th></tr></thead>
<tbody><tr><td>

<img width="720" alt="CleanShot 2022-09-14 at 07 18 08@2x" src="https://user-images.githubusercontent.com/15199/190128760-86b542bc-2bd7-4979-963d-84e378027a67.png">

</td><td>

<img width="882" alt="CleanShot 2022-09-14 at 07 22 25@2x" src="https://user-images.githubusercontent.com/15199/190129359-4735a7f9-6d0d-4949-915e-1dbb69a9884b.png">


</td></tr></tbody>
</table>

**Comment by henningmu** [comment]
@gnapse this issue was motivated by work on this (internal) issue: https://github.com/Doist/Issues/issues/7617 – the new alert designs are more lightweight, making me question the usefulness of Notice even more 👍

**Comment by gnapse** [comment]
Indeed, if the design team decides that we can replace all our uses of the standalone instances of `<Notice />` with these new lighter alerts, then I'm ok with it. cc @Doist/design-product-hero 

To be clear, it would not affect the notice-like messages embedded as part of the field component, but it would mean that we will no longer be able to render these borderless and background-less alerts (that's basically what notices are: borderless and backgroundless alerts that also cannot show any extra UI elements inside it such as buttons aligned to the right).

If we're ok with that, then we deprecate it, and eventually remove it. The less we have to maintain, the better.

**Comment by anaf** [comment]
I think that could work. 
But before confirming it, do we have an easy way to find all the instances ```Notice```, just so I can make sure?

**Comment by gnapse** [comment]
@anaf sorry for the very late response. I was reviewing old issues in Reactist and realized this question slipped through the cracks. Here's a list of the current uses of `<Notice />` in Todoist:

<table>
  <tr>
    <td>(1) Project export template as link: showing the error message.</td>
    <td><img width="466" src="https://github.com/Doist/reactist/assets/15199/38c37b8c-397d-45dc-9c5c-96e933d0b913"></td>
  </tr>
  <tr>
    <td>(2) Reminders picker, showing error when date entered is invalid:</td>
    <td><img width="350" src="https://github.com/Doist/reactist/assets/15199/563e67fe-86b1-41a5-a3be-449ea6d3964d"></td>
  </tr>
  <tr>
    <td>(3) Account settings, when enabling two-factor auth results in an error:</td>
    <td><img width="335" src="https://github.com/Doist/reactist/assets/15199/376fc617-7978-4466-a95e-abb2c3c1d3d8"></td>
  </tr>
  <tr>
    <td>(4) Similar to the previous use case. If something goes wrong inside the two-factor auth settings.</td>
    <td><img width="283" src="https://github.com/Doist/reactist/assets/15199/679d9a5d-12f0-4095-a021-af588673252e"></td>
  </tr>
  <tr>
    <td>(5) When an error occurs while trying to disconnect from an auth provider</td>
    <td><img width="466" src="https://github.com/Doist/reactist/assets/15199/9cb3985f-8dbc-4445-9eac-0d6853d572ed"></td>
  </tr>
  <tr>
    <td>(6) To show the error when trying to delete your account:</td>
    <td><img width="466" src="https://github.com/Doist/reactist/assets/15199/901233af-9f94-4d8b-841e-e57ea26a1865"></td>
  </tr>
<tr>
<td>(7) Showing an error when it fails to add or update the payment method. Or when trying to remove a payment method <br/>(Screenshot and context for that last one may be different, but it was not easily reproducible in my test account in the short term. Let me know if you really need it.)</td>
<td><img width="745" alt="image" src="https://github.com/Doist/reactist/assets/15199/fb381c4c-795a-4815-84ec-023c930b753c">
</td>
</tr>
  <tr>
    <td>(8) When entering an invalid email while inviting people to a workspace</td>
    <td><img width="472" alt="CleanShot 2023-05-29 at 16 36 12@2x" src="https://github.com/Doist/reactist/assets/15199/7c77376e-28e2-410c-9fa1-a410aaec8687"></td>
  </tr>
</table>
